### ISSUE TITLE
Doesn't Download and decode progressive JPEGs on iOS.

### ISSUE THREAD

**Issue by DavidPhillipOster** [contentful-labs/Concorde#18]
The title of this github repository is "Download and decode progressive JPEGs on iOS." But it doesn't do the very thing that Progressive JPEG was invented to do:

**Progressively** download and decode JPEGs on iOS

It misuses NSURLSession to grab ALL the data (a process that takes 99% of the total time) then decodes it.

Please rewrite to use NSURLSession in stream mode, so that is passed to your CGDataProvider in bursts, decoding what you can and writing it to the screen as the data comes in.

In CCBufferedImageDecoder , you create your CGDataProvider with an NSData, but by the time you have the NSData, 99% of the time of fetching and displaying the image has already elapsed, making progressive jpeg decoding pointless.

**Comment by oferRounds** [comment]
@DavidPhillipOster did you find any solution to this? Seeing it as well

**Comment by DavidPhillipOster** [comment]
no, I've outlined the solution: use a  CGDataProvider that in turn is driven from a NSURLSession in stream mode, but I haven't implemented it (I haven't needed it), and this project has seen no check ins in years.
